src/ripple/app/ledger/impl/InboundLedgers.cpp

```diff
@@ -67,26 +66,26 @@ class InboundLedgersImp
                 auto it = mLedgers.find (hash);
                 if (it != mLedgers.end ())
                 {
-                    // Don't touch failed acquires so they can expire
-                    if (! it->second->isFailed ())
-                    {
-                        it->second->update (seq);
-                        if (it->second->isComplete ())
-                            ret = it->second->getLedger ();
-                    }
+                    inbound = it->second;
 
+                    // If the acquisition failed, don't mark the item as
+                    // recently accessed so that it can expire.
+                    if (! inbound->isFailed ())
+                        it->second->update (seq);
```

Comment:
Why not `inbound->update (seq);` - provably the same because of line 69 but the logic is a tiny bit clearer...